### PR TITLE
Spike versioning docs

### PR DESCRIPTION
**Pull request by laurakalbag**
This spike adds Docusaurus versioning to our docs.

The “current” version is kept in the root directory. “Next” and “previous” versions are kept in versioned folders. There are also version-specific navigation sidebars.

## Upsides
- each version is clearly labelled (old and future versions all have a banner at the top of their pages)
- easy to switch between versions from top bar
- you can have different pages for each version, navigated from the left navigation sidebar and the next/prev links
- you can also have equivalent pages for each version, which can be switched between from the top bar
- the site remembers the person’s last version and takes them there next visit

## Downsides:
1. You have to use relative links to link to other docs pages.
2. When the content is the same across two or more versions, we have to maintain identical versions of the same page. This is just how Docusaurus does versioning now. In their last major version, they actually did have a fallback method, but I can’t even find a workaround for doing this now. We would only maintain the current and next versions of each page (nobody would expect previous versions to have up-to-date content) but this is a lot of manual effort for us, and would make contributions really difficult to both contribute and review.

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -49,6 +49,20 @@ const config = {
           // Remove this to remove the "edit this page" links.
           editUrl: 'https://github.com/statelyai/docs/tree/main/',
 
+          lastVersion: 'current',
+          includeCurrentVersion: true,
+          versions: {
+            current: {
+              label: 'XState v4',
+              path: 'xstate-v-4',
```
Comment (mellson) [inline docusaurus.config.js:57]: Try removing this `path` entirely. Then I think it will default to `/` for the current version 🤞🏻

Comment (laurakalbag) [reply docusaurus.config.js:57]: You’re a genius! That worked!

Comment (laurakalbag) [reply docusaurus.config.js:57]: (Thank you!)

**Comment by davidkpiano** [comment]
Quick note: might need some `white-space: nowrap` here

CleanShot 2023-03-21 at 23 55 54

**Comment by laurakalbag** [comment]
> Quick note: might need some `white-space: nowrap` here
> 
> CleanShot 2023-03-21 at 23 55 54

Oh yes! I totally didn’t do any styling there yet… I might need to tackle some overcrowding in the top bar when we merge with the blog spike too…

**Review by davidkpiano — APPROVED** [review]
Preemptively approved, doesn't have to go in yet (we can consider this the feature branch)

**Pull request closed without merging**